state/execution.go

```diff
@@ -73,21 +80,28 @@ func execBlockOnProxyApp(eventCache types.Fireable, proxyAppConn proxy.AppConnCo
 			// Blocks may include invalid txs.
 			// reqDeliverTx := req.(abci.RequestDeliverTx)
 			txError := ""
-			apTx := r.DeliverTx
-			if apTx.Code == abci.CodeType_OK {
-				validTxs += 1
+			txResult := r.DeliverTx
+			if txResult.Code == abci.CodeType_OK {
+				validTxs++
 			} else {
-				log.Debug("Invalid tx", "code", r.DeliverTx.Code, "log", r.DeliverTx.Log)
-				invalidTxs += 1
-				txError = apTx.Code.String()
+				log.Debug("Invalid tx", "code", txResult.Code, "log", txResult.Log)
+				invalidTxs++
+				txError = txResult.Code.String()
+			}
+
+			tx := types.Tx(req.GetDeliverTx().Tx)
+			index, ok := txHashToIndexMap[string(tx.Hash())]
+			if ok {
```

Comment:
how could we have not ok? is this map really necessary? these callbacks should be called in order so we should just be able to use a simple counter. Or am I missing something?